File: src/main/java/com/anod/appwatcher/search/ResultsAdapterDetails.java
package com.anod.appwatcher.search;

import android.content.Context;

import com.anod.appwatcher.market.DetailsEndpoint;
import com.anod.appwatcher.model.WatchAppList;
import com.google.android.finsky.api.model.Document;

/**
 * @author algavris
 * @date 26/08/2016.
 */

public class ResultsAdapterDetails extends ResultsAdapter {
    private final DetailsEndpoint mDetailsEndpoint;

    public ResultsAdapterDetails(Context context, DetailsEndpoint detailsEndpoint, WatchAppList newAppHandler) {
        super(context, newAppHandler);
        mDetailsEndpoint = detailsEndpoint;
    }

    @Override
    public Document getDocument(int position) {
        return mDetailsEndpoint.getDocument();
    }

    @Override
    public int getItemCount() {
        return mDetailsEndpoint.getDocument() != null ? 1 : 0;
    }
}

File: src/main/java/com/anod/appwatcher/search/ResultsAdapterDetails.kt
package com.anod.appwatcher.search

import android.content.Context

import finsky.api.model.Document

/**
 * @author Alex Gavrishev
 * *
 * @date 26/08/2016.
 */

class ResultsAdapterDetails(context: Context, viewModel: SearchViewModel)
    : ResultsAdapter(context, viewModel) {

    private val endpoint = viewModel.endpointDetails

    override fun document(position: Int): Document {
        return endpoint.document!!
    }

    override fun getItemCount(): Int {
        return if (endpoint.document != null) 1 else 0
    }
}
